geonode/base/api/serializers.py

```diff
@@ -376,7 +376,7 @@ class Meta:
 
     def to_representation(self, instance):
         data = []
-        if ResourceBase.objects.filter(pk=instance).count() == 1:
+        if not self.context['request'].user.is_anonymous and ResourceBase.objects.filter(pk=instance).count() == 1:
```

Comment:
@giohappy What do you think about using a get on `self.context.get('request')`, and also ensuring we have a request before calling a user. In case the serializer is used in representing data without a request involved.

Reply:
Done, thanks @marthamareal